execution-plane/arcon/src/data/buffer.rs

```diff
@@ -0,0 +1,295 @@
+// Copyright (c) 2020, KTH Royal Institute of Technology.
+// SPDX-License-Identifier: AGPL-3.0-only
+
+use crate::allocator::{AllocId, AllocResult, ArconAllocator};
+use arcon_error::*;
+use crossbeam_utils::CachePadded;
+use std::sync::atomic::{AtomicBool, Ordering};
+use std::sync::{Arc, Mutex};
+
+/// A reusable buffer allocated through [ArconAllocator]
+///
+/// Assumes a Single-writer, single-reader setup.
+#[derive(Debug)]
+pub struct EventBuffer<T> {
```

Comment:
The `Drop` impls of `T`s contained within are never called, right? Might want to document that.

Reply:
Potentially related: http://cglab.ca/~abeinges/blah/everyone-poops/

Reply:
This would only be a problem if the EventBuffer is dropped before the BufferReader is consumed. As the BufferReader.into_iter() call will consume all T's and take ownership and thus drop them once they go out of scope.